packages/itwinui-css/src/button/button-group.scss

```diff
@@ -63,43 +59,38 @@
       }
     }
 
-    // Add rounded corners to first and last items
-    &:where(:first-child) {
-      #{$childBorderSelector} {
-        border-start-start-radius: var(--iui-border-radius-1);
+    // Add rounded corners to first item
+    &:where(:nth-child(1 of :where(.iui-field, .iui-input-flex-container))) {
+      border-start-start-radius: var(--iui-border-radius-1);
 
-        @if $orientation == 'horizontal' {
-          border-end-start-radius: var(--iui-border-radius-1);
-        } @else {
-          border-start-end-radius: var(--iui-border-radius-1);
-        }
+      @if $orientation == 'horizontal' {
+        border-end-start-radius: var(--iui-border-radius-1);
+      } @else {
+        border-start-end-radius: var(--iui-border-radius-1);
       }
     }
 
-    &:where(:last-child) {
-      #{$childBorderSelector} {
-        border-end-end-radius: var(--iui-border-radius-1);
+    // Add rounded corners to last item
+    &:where(:nth-last-child(1 of :where(.iui-field, .iui-input-flex-container))) {
+      border-end-end-radius: var(--iui-border-radius-1);
 
-        @if $orientation == 'horizontal' {
-          border-start-end-radius: var(--iui-border-radius-1);
-        } @else {
-          border-end-start-radius: var(--iui-border-radius-1);
-        }
+      @if $orientation == 'horizontal' {
+        border-start-end-radius: var(--iui-border-radius-1);
+      } @else {
+        border-end-start-radius: var(--iui-border-radius-1);
       }
     }
 
     // Use negative margin to make adjacent borders overlap
-    + * {
+    ~ :where(.iui-field, .iui-input-flex-container) {
       @if $orientation == 'horizontal' {
         margin-inline-start: -1px;
       } @else {
         margin-block-start: -1px;
       }
-    }
 
-    // Add slight border between disabled items
-    &:where(:not(:first-child):not([data-iui-variant='borderless'])) {
-      #{$disabledSelector} {
+      // Add slight border between disabled items
+      &:where(:disabled, [aria-disabled='true']):where(:not([data-iui-variant='borderless'])) {
```

Comment:
Should we continue to exclude the first child for the disabled border?

Reply:
no, this is chained to the `~` selector, which means it will never match the first child